index.js

```diff
@@ -81,19 +92,23 @@ let Rosnodejs = {
    * @return {Promise} resolved when connection to master is established
    */
   initNode(nodeName, options) {
+    options = options || {};
+    if (options.anonymous) {
+      nodeName = _anonymizeNodeName(nodeName);
+      console.log('anonymous node name ' + nodeName);
+    }
+
     nodeName = _validateNodeName(nodeName);
 
     if (rosNode !== null) {
       if (nodeName === rosNode.getNodeName()) {
         return Promise.resolve(this.getNodeHandle());
       }
       // else
-      throw new Error('Unable to initialize node [' + nodeName + '] - node [' 
+      throw new Error('Unable to initialize node [' + nodeName + '] - node ['
                       + rosNode.getNodeName() + '] already exists');
     }
 
-    // FIXME: validate nodeName -- MUST START WITH '/'
```

Comment:
I meant to remove it when I added the validation that exists. There is, admittedly, a lot more name validation/mapping that we need to do, though this exists for most ROS names and not specifically for node names.